src/reflect/value.go

```diff
@@ -1065,9 +1065,52 @@ func (v Value) FieldByName(name string) Value {
 	panic("unimplemented: (reflect.Value).FieldByName()")
 }
 
+//go:linkname hashmapMake runtime.hashmapMakeUnsafePointer
+func hashmapMake(keySize, valueSize uintptr, sizeHint uintptr, alg uint8) unsafe.Pointer
+
+// MakeMapWithSize creates a new map with the specified type and initial space
+// for approximately n elements.
+func MakeMapWithSize(typ Type, n int) Value {
+
+	const (
+		hashmapAlgorithmBinary uint8 = iota
+		hashmapAlgorithmString
+		hashmapAlgorithmInterface
+	)
```

Comment:
It's kind of unfortunate to have this duplicated.
I guess this could be fixed using an internal package that defines these constants? Not sure it's worth the trouble (and probably better to do in a separate PR if you think it's worth the deduplication).

Reply:
Yeah.  The runtime needs them, the reflect code needs them, and the compiler needs them.